genume/constants.py

```diff
@@ -39,7 +39,7 @@
 
 # Limit the maximum number of child processes to have running at any given time.
 # This limit is imposed by each instance of Registry separately.
-SCRIPTS_MAX_MULTI_DISPATCH = max(min(multiprocessing.cpu_count(), 32), 4)
+SCRIPTS_MAX_MULTI_DISPATCH = max(min(multiprocessing.cpu_count(), 32), 4) * 20
```

Comment:
If this variable is set to a large value, it may cause some systems to run out of resources (mem, pid, tid), as its effects are similar to a fork bomb.

Also welcome back @MaanooAk.

Reply:
20 processes per core is nothing, a bomb would be over 1000. The consept is to let OS handle the parallel execution and not genome, so I would prefer to have `* Infinity` but considering what you are talking about a made it 20.

Reply:
With the help of [cgmemtime](https://github.com/gsauthof/cgmemtime), I calculated that the average memory usage of the current 18 scripts is ~6Mib each. This gives us a 108Mib of peak memory usage, or 2.8% of my system's ram.

However 18 processes is quite low, compared to the minimum of `4*20=80` slots. In this case the peak memory usage would be ~480Mib, or 12% of my total ram.

As I have to always work under low memory, I hate people who waste ram for no reason.

Reply:
Please take a look at 78bacfe165ae60c5309e6486d73af167103f167e.

Reply:
_[362 days later]_ 
@Sima214 Don't really remember what we were doing, but I tested the dev branch seems good, I would suggest to merge it